osbs/build/build_request.py

```diff
@@ -465,21 +468,72 @@ def set_kerberos_auth(self, plugins):
             self.dj.dock_json_set_arg(phase, plugin, 'koji_principal', krb_principal)
             self.dj.dock_json_set_arg(phase, plugin, 'koji_keytab', krb_keytab)
 
-    def set_reactor_config_map(self):
-        if not self.spec.reactor_config_map.value:
+    def set_reactor_config(self):
+        if not self.spec.reactor_config_map.value and not self.spec.reactor_config_override.value:
             return
         custom = self.template['spec']['strategy']['customStrategy']
 
-        reactor_config_map = {
-            'name': 'REACTOR_CONFIG',
-            'valueFrom': {
-                'configMapKeyRef': {
-                    'name': self.spec.reactor_config_map.value,
-                    'key': 'config.yaml'
+        if self.spec.reactor_config_override.value:
+            reactor_config = {
+                'name': 'REACTOR_CONFIG',
+                'value': yaml.dump(self.spec.reactor_config_override.value)
+            }
+        elif self.spec.reactor_config_map.value:
+            reactor_config = {
+                'name': 'REACTOR_CONFIG',
+                'valueFrom': {
+                    'configMapKeyRef': {
+                        'name': self.spec.reactor_config_map.value,
+                        'key': 'config.yaml'
+                    }
                 }
             }
-        }
-        custom['env'].append(reactor_config_map)
+
+        custom['env'].append(reactor_config)
+
+    def set_required_secrets(self):
+        """
+        Sets required secrets
+        """
+        if not self.spec.reactor_config_map.value and not self.spec.reactor_config_override.value:
+            return
+
+        req_secrets_key = 'required_secrets'
+        required_secrets = []
+        config_map = None
+        if self.spec.reactor_config_override.value:
+            data = self.spec.reactor_config_override.value
+            required_secrets = data.get(req_secrets_key)
+        elif self.spec.reactor_config_map.value:
+            config_map = self.osbs_api.get_config_map(self.spec.reactor_config_map.value)
+            required_secrets = config_map.get_data_by_key('config.yaml').get(req_secrets_key, [])
+
+        if self.spec.platforms.value is not None and self.spec.reactor_config_map.value:
+            if not config_map:
+                config_map = self.osbs_api.get_config_map(self.spec.reactor_config_map.value)
+            required_secrets +=\
+                config_map.get_data_by_key('config.yaml').get('worker_token_secrets', [])
+
+        if not required_secrets:
+            return
+
+        secrets = self.template['spec']['strategy']['customStrategy'].setdefault('secrets', [])
+        existing = set(secret_mount['secretSource']['name'] for secret_mount in secrets)
+
+        for secret in set(required_secrets):
```

Comment:
Could exclude `existing` before iterating:
```python
for secret in set(required_secrets) - existing:
  ...
```

Reply:
so we don't want to report debug that secret is already set?

Reply:
I don't see a lot of value in reporting that, but you could log `existing` before entering the loop.

Reply:
ok I'll report intersection then